extensions/sns/src/main.rs

```diff
@@ -68,49 +76,52 @@ enum SubCommand {
 }
 
 /// Executes `dfx sns` and its subcommands.
-fn main() -> anyhow::Result<()> {
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
     let opts = SnsOpts::parse();
 
-    match opts.subcmd {
-        SubCommand::DeployTestflight(args) => {
-            deploy_testflight(args);
-            Ok(())
-        }
-        SubCommand::InitConfigFile(args) => {
-            init_config_file::exec(args);
-            Ok(())
-        }
-        SubCommand::PrepareCanisters(args) => {
-            prepare_canisters::exec(args);
-            Ok(())
-        }
-        SubCommand::Propose(args) => {
-            propose::exec(args);
-            Ok(())
-        }
+    let subcommand: SnsLibSubCommand = match opts.subcmd {
+        SubCommand::DeployTestflight(args) => SnsLibSubCommand::DeployTestflight(args),
+        SubCommand::InitConfigFile(args) => SnsLibSubCommand::InitConfigFile(args),
+        SubCommand::PrepareCanisters(args) => SnsLibSubCommand::PrepareCanisters(args),
+        SubCommand::Propose(args) => SnsLibSubCommand::Propose(args),
         SubCommand::NeuronIdToCandidSubaccount(args) => {
-            neuron_id_to_candid_subaccount::exec(args);
-            Ok(())
+            SnsLibSubCommand::NeuronIdToCandidSubaccount(args)
         }
+        SubCommand::AddSnsWasmForTests(args) => SnsLibSubCommand::AddSnsWasmForTests(args),
+        SubCommand::List(args) => SnsLibSubCommand::List(args),
 
         SubCommand::Import(v) => {
             let dfx_cache_path = &opts.dfx_cache_path.ok_or_else(|| {
                 anyhow::Error::msg(
                     "Missing path to dfx cache. Pass it as CLI argument: `--dfx-cache-path=PATH`",
                 )
             })?;
-            commands::import::exec(v, dfx_cache_path)
+            return commands::import::exec(v, dfx_cache_path);
         }
         SubCommand::Download(v) => {
             let dfx_cache_path = &opts.dfx_cache_path.ok_or_else(|| {
                 anyhow::Error::msg(
                     "Missing path to dfx cache. Pass it as CLI argument: `--dfx-cache-path=PATH`",
                 )
             })?;
-            commands::download::exec(v, dfx_cache_path)
+            return commands::download::exec(v, dfx_cache_path);
+        }
+    };
+
+    let agent = utils::get_mainnet_agent()?;
+
+    match subcommand {
+        SnsLibSubCommand::DeployTestflight(args) => deploy_testflight(args),
+        SnsLibSubCommand::InitConfigFile(args) => init_config_file::exec(args),
+        SnsLibSubCommand::PrepareCanisters(args) => prepare_canisters::exec(args),
+        SnsLibSubCommand::Propose(args) => propose::exec(args),
+        SnsLibSubCommand::NeuronIdToCandidSubaccount(args) => {
+            neuron_id_to_candid_subaccount::exec(args)
         }
-    }?;
-    Ok(())
+        SnsLibSubCommand::AddSnsWasmForTests(args) => add_sns_wasm_for_tests(args),
+        SnsLibSubCommand::List(args) => list::exec(args, &agent).await,
```

Comment:
NCR

Would be nice if all these looked the same. I.e.

```
X::Y(y) => y::exec(y),
```

Even better (but more work, and therefore, not recommended now, but next time):

```
subcommand.unwrap().exec()
```

where unwrap returns a `Box<dyn Exec>`, and `Exec` is a new trait that requires `fn exec(&self) -> Result<...>;`, and all the args types implement it.

Reply:
Unfortunately this difference is an annoying consequence of the scoping of `add_sns_wasm_for_tests`, which can't be fixed in this PR as it's determined by the monorepo :(